lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -1069,6 +1506,42 @@ uint64_t PatchEntryPointMutate::pushFixedShaderArgTys(SmallVectorImpl<Type *> &a
   return inRegMask;
 }
 
+// =====================================================================================================================
+// Get UserDataUsage struct for the merged shader stage that contains the given shader stage
+//
+// @param stage : Shader stage
+PatchEntryPointMutate::UserDataUsage *PatchEntryPointMutate::getUserDataUsage(ShaderStage stage) {
+  stage = getMergedShaderStage(stage);
+  m_userDataUsage.resize(std::max(m_userDataUsage.size(), static_cast<size_t>(stage) + 1));
+  if (!m_userDataUsage[stage])
+    m_userDataUsage[stage] = std::make_unique<UserDataUsage>();
+  return &*m_userDataUsage[stage];
+}
+
+// =====================================================================================================================
+// Get the shader stage that the given shader stage is merged into.
+// For GFX9+:
+// VS -> TCS (if it exists)
+// VS -> GS (if it exists)
+// TES -> GS (if it exists)
+ShaderStage PatchEntryPointMutate::getMergedShaderStage(ShaderStage stage) const {
+  if (m_pipelineState->getTargetInfo().getGfxIpVersion().major >= 9) {
+    switch (stage) {
+    case ShaderStageVertex:
+      if (m_pipelineState->hasShaderStage(ShaderStageTessControl))
+        return ShaderStageTessControl;
+      LLVM_FALLTHROUGH;
```

Comment:
This is very interesting. LLVM define such marker. Previously, PAL style requires a comment // Fall through for such use. But I think it is OK to use this marker. Better to write it to coding style doc

Reply:
I get a warning (error with -Werror) if I do not use it.

I just checked, and I was surprised that it is not mentioned in the LLVM coding standards doc. So, yes, we should add it to our doc.